packages/imperative/src/rest/src/session/doc/IOptionsForAddConnProps.ts

```diff
@@ -51,7 +50,19 @@ export interface IOptionsForAddConnProps {
      * Specifies a list of authentication properties, and what they should override.
      * If one of these properties is available on the session, do not prompt for the other property.
      */
-    propertyOverrides?: IOverridePromptConnProps[];
+    propertyOverrides?: IOverridePromptConnProps<SessCfgType>[];
+
+    /**
+     * Allows passing additional properties for which to prompt.
+     * Used in cases of an incorrect or missing key passphrase.
+     */
+    propsToPromptFor?:
+    {
+        name: keyof SessCfgType & string,
+        secure?: boolean,
+        description?: string,
+        isGivenValueValid?: (givenValue: string) => boolean
+    }[];
```

Comment:
Could this be defined as an interface/type, or do we feel that defining it as one would be redundant?
If we only intend on using the typed object within this interface, then we could probably leave it as-is.

Reply:
Yeah, I do see some value in cleaning this up a little.
```suggestion
    IPropsToPromptFor<SessCfgType>[];
```

And then import...

```typescript
import { IPropsToPromptFor } from "./IPropsToPropmtFor";
```

Which could be similar to `IOverridePromptConnProps` but with the following contents...

```typescript
export interface IPropsToPromptFor <SessCfgType extends ISession=ISession> {
  name: keyof SessCfgType & string,
  secure?: boolean,
  description?: string,
  isGivenValueValid?: (givenValue: string) => boolean
}
```